base/_utils.py

```diff
@@ -432,21 +432,24 @@ class matcher(object):
 
     def __init__(self):
         self.__predicate__ = {}
-    def __attrib__(self, *attribute):
-        if not attribute:
-            return lambda n: n
-        res = [(operator.attrgetter(a) if isinstance(a, six.string_types) else a) for a in attribute]
-        return lambda o: tuple(x(o) for x in res) if len(res) > 1 else res[0](o)
+    def __attrib__(self, *attributes):
+        if not attributes:
+            return lambda item: item
+        res = [(operator.attrgetter(callable_or_attribute) if isinstance(callable_or_attribute, six.string_types) else callable_or_attribute) for callable_or_attribute in attributes]
+        return fcompose(*res) if len(res) > 0 else res[0]
     def attribute(self, type, *attribute):
         attr = self.__attrib__(*attribute)
-        self.__predicate__[type] = lambda v: fcompose(attr, functools.partial(functools.partial(operator.eq, v)))
+        self.__predicate__[type] = lambda target: fcompose(attr, functools.partial(functools.partial(operator.eq, target)))
```

Comment:
This should probably be combinator based...but I really don't care to fuck with it right now.